Expensify/App
# [No QA] Stop sending `undefined` request params to the API

### Details

It is extremely hard to tell why someone was booted from NewDot because we are claiming that they are missing a 2FA token. In reality, we are sending a STRING of `'undefined'` to Web-Expensify and that is a truthy value. Solution is to filter out these kinds of parameters because Web-E is surely going to do some unexpected things.

See linked issue for more details.

### Fixed Issues
$ https://github.com/Expensify/App/issues/8637

### Tests
1. Create a new account in expensify.com.dev
2. Jump to NewDot via OldDot without validating this account
3. Bork your authToken somehow like this
```diff
diff --git a/src/libs/Network/NetworkStore.js b/src/libs/Network/NetworkStore.js
index c0973fa50..f379006a8 100644
--- a/src/libs/Network/NetworkStore.js
+++ b/src/libs/Network/NetworkStore.js
@@ -11,6 +11,8 @@ let hasReadRequiredData = false;
 let isAuthenticating = false;
 let isOffline = false;

+window.breakAuthToken = () => authToken = 'pizza';
+
```
4. Do something that triggers an API request.
5. Verify that you only sent good values to the API and not `undefined` ones i.e. `twoFactorCode` should not be present at all
![2022-04-15_13-33-16](https://user-images.githubusercontent.com/32969087/163652551-654333c0-73e9-42bb-a664-90c04995258a.png)
6. You will be logged out (expected)
7. Check logs and verify that the reason you were logged out is not related to a 2FA code and is clearly related to the lack of account validation.
